plugins/by-name/avante/default.nix

```diff
@@ -0,0 +1,95 @@
+{ lib, pkgs, ... }:
+let
+  inherit (lib.nixvim) defaultNullOpts mkNullOrOption' toLuaObject;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "avante";
+  originalName = "avante.nvim";
+  package = "avante-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  # TODO: remove when XXX reaches `nixos-unstable`
+  extraPlugins = with pkgs.vimPlugins; [
+    dressing-nvim
+    nui-nvim
+    plenary-nvim
+  ];
+
+  settingsOptions = {
+    provider = defaultNullOpts.mkStr "claude" ''
+      The LLM provider (`"claude"`, `"openai"`, `"azure"`, ...)
+    '';
+
+    auto_suggestions_frequency = defaultNullOpts.mkStr "claude" ''
+      The provider for automatic suggestions.
+
+      Since auto-suggestions are a high-frequency operation and therefore expensive, it is
+      recommended to specify an inexpensive provider or even a free provider: `"copilot"`.
+    '';
+
+    mappings = mkNullOrOption' {
+      type = with lib.types; attrsOf (either str (attrsOf str));
+      example = {
+        diff = {
+          ours = "co";
+          theirs = "ct";
+          none = "c0";
+          both = "cb";
+          next = "]x";
+          prev = "[x";
+        };
+        jump = {
+          next = "]]";
+          prev = "[[";
+        };
+      };
+    };
+  };
+
+  settingsExample = {
+    provider = "claude";
+    claude = {
+      endpoint = "https://api.anthropic.com";
+      model = "claude-3-5-sonnet-20240620";
+      temperature = 0;
+      max_tokens = 4096;
+    };
+    mappings = {
+      diff = {
+        ours = "co";
+        theirs = "ct";
+        none = "c0";
+        both = "cb";
+        next = "]x";
+        prev = "[x";
+      };
+    };
+    hints.enabled = true;
+    windows = {
+      wrap = true;
+      width = 30;
+      sidebar_header = {
+        align = "center";
+        rounded = true;
+      };
+    };
+    highlights.diff = {
+      current = "DiffText";
+      incoming = "DiffAdd";
+    };
+    diff = {
+      debug = false;
+      autojump = true;
+      list_opener = "copen";
+    };
+  };
+
+  callSetup = false;
+  extraConfig = cfg: {
+    extraConfigLua = ''
+      require('avante_lib').load()
+      require('avante').setup (${toLuaObject cfg.settings})
```

Comment:
This is the kind of stuff that makes me want https://github.com/nix-community/nixvim/pull/1876 available. @traxys are you still interested in getting that in?

Reply:
Yeah, need to finish addressing the feedback on it

Reply:
#1876 has been merged

Reply:
In this case, I just add `require('avante_lib').load()` to `luaConfig.pre` ?

Reply:
done